libfs/fs.go

```diff
@@ -366,15 +387,22 @@ func (fs *FS) Join(elem ...string) string {
 
 // TempFile implements the billy.Filesystem interface for FS.
 func (fs *FS) TempFile(dir, prefix string) (billy.File, error) {
-	// We'd have to turn off journaling to support TempFile fully.
-	return nil, errors.New("TempFile is not currently supported")
+	// We'd have to turn off journaling to support TempFile perfectly,
+	// but the given uniq ID and a random number should be good
+	// enough.  Especially since most users will end up renaming the
+	// temp file before journal flushing even happens.
+	r := rand.Int63()
```

Comment:
I didn't know the answer, so I looked up how Go's `ioutil.TempFile` does it, and it turns out it's just an integer too: https://golang.org/src/io/ioutil/tempfile.go?s=#L26

So I guess it's fine. Although do we call `rand.Seed()` somewhere? Otherwise it's always the same sequence and we'd end up with the same file name. Also I'm curious why `rand` instead of `crypto/rand`? Is it performance concern too?